### ISSUE TITLE
NameError: name 'httpx' is not defined

### ISSUE THREAD

**Issue by ydc-0** [python-gitlab/python-gitlab#2981]
## Description of the problem, including code/CLI snippet

Failed to `import gitlab`
```
Python 3.11.4 (main, Aug 10 2023, 17:15:32) [GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gitlab
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/chenyudong/.pyenv/pyenv/versions/3.11.4/lib/python3.11/site-packages/gitlab/__init__.py", line 30, in <module>
    from gitlab.client import Gitlab, GitlabList, GraphQL  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/chenyudong/.pyenv/pyenv/versions/3.11.4/lib/python3.11/site-packages/gitlab/client.py", line 1279, in <module>
    class GraphQL:
  File "/home/chenyudong/.pyenv/pyenv/versions/3.11.4/lib/python3.11/site-packages/gitlab/client.py", line 1286, in GraphQL
    client: Optional[httpx.Client] = None,
                     ^^^^^
NameError: name 'httpx' is not defined
```

## Specifications

  - python-gitlab version: 4.11.0
  - API version you are using (v3/v4):
  - Gitlab server version (or gitlab.com):

**Comment by nejch** [comment]
Thanks for the report @ydc-0 we'll try to fix the regression ASAP.

**Comment by hfudev** [comment]
@nejch or shall the version be yanked?

**Comment by nejch** [comment]
@ydc-0 we probably will but will still try to get the fix released ASAP.

**Comment by nejch** [comment]
If you like you can check the fix with `pip install git+https://github.com/python-gitlab/python-gitlab.git@9e0f8f307d70b27acf5f596838e08f491063d396` just in case - this worked for me on 3.11 and 3.12.

**Comment by ydc-0** [comment]
@nejch thanks for the quick fix.

 It works on python 3.11 and python 3.8

```
➜  ~ python
Python 3.8.17 (default, Sep 13 2024, 21:19:39) 
[GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gitlab

```

**Comment by nejch** [comment]
Thanks for the confirmation @ydc-0.

For anyone else reading this, v4.11.1 is now out on pypi with the fix: https://pypi.org/project/python-gitlab/4.11.1/

We yanked 4.11.0 since projects may be depending on python-gitlab and do not properly pin it, so it was best to remove it. This actually happened here when we tried to re-release using the python-semantic-release GH action which in turns uses python-gitlab without a lockfile.
